qa/rpc-tests/prioritisetransaction.py

```diff
@@ -38,95 +103,127 @@ def run_test (self):
+            assert_equal(priority_success, True)
+
+            # Check that priority_tx_0 was not included by priority through getblocktemplate()
+            # (not updated because no new txns)
+            block_template = self.nodes[0].getblocktemplate()
+            tx_0_included_by_priority, tx_0_included = self.check_if_included_by_priority(priority_tx_0, last_tx_to_include_by_priority_0, block_template)
+            assert_equal(tx_0_included_by_priority, False)
 
-        assert(in_block_template)
-
-        # Node 1 doesn't get the next block, so this *shouldn't* be mined despite being prioritized on node 1
-        priority_tx_1 = self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 0.1)
-        self.nodes[1].prioritisetransaction(priority_tx_1, 1000, int(3 * base_fee * COIN))
-
-        # Mine block on node 0
-        blk_hash = self.nodes[0].generate(1)
-        block = self.nodes[0].getblock(blk_hash[0])
-        self.sync_all()
-
-        # Check that priority_tx_0 was mined
-        mempool = self.nodes[0].getrawmempool()
-        assert_equal(priority_tx_0 in block['tx'], True)
-        assert_equal(priority_tx_0 in mempool, False)
-
-        # Check that priority_tx_1 was not mined
-        assert_equal(priority_tx_1 in mempool, True)
-        assert_equal(priority_tx_1 in block['tx'], False)
+            # Sending a new transaction will make getblocktemplate refresh within 10s
+            self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1)
+            self.sync_all()
 
-        # Mine a block on node 1 and sync
-        blk_hash_1 = self.nodes[1].generate(1)
-        block_1 = self.nodes[1].getblock(blk_hash_1[0])
-        self.sync_all()
+            # Check that priority_tx_0 was not included by priority through getblocktemplate()
+            # (too soon)
+            block_template = self.nodes[0].getblocktemplate()
+            tx_0_included_by_priority, tx_0_included = self.check_if_included_by_priority(priority_tx_0, last_tx_to_include_by_priority_0, block_template)
+            assert_equal(tx_0_included_by_priority, False)
+
+            # Check that priority_tx_0 was by priority through getblocktemplate()
+            # getblocktemplate() will refresh after 1 min, or after 10 sec if new transaction is added to mempool
+            # Mempool is probed every 10 seconds. We'll give getblocktemplate() a maximum of {timeout_seconds} seconds to refresh
+            start = time.time()
+            in_block_template = False
+            while in_block_template == False:
+                block_template = self.nodes[0].getblocktemplate()
+                tx_0_included_by_priority, tx_0_included = self.check_if_included_by_priority(priority_tx_0, last_tx_to_include_by_priority_0, block_template)
+                if (round == 0 and tx_0_included_by_priority):
+                    break
+                elif (round == 1 and tx_0_included):
+                    break
```

Comment:
Just a style thing, but I find it a bit more readable and debug friendly (e.g. at round 0, if tx is included but not by priority). How do you like this suggestion, which also replaces lines 172-176?
```suggestion
                if (tx_0_included):
                    assert_equal(round == 0, tx_0_included_by_priority)
                    assert_equal(round == 1, !tx_0_included_by_priority)
                    break
```

Reply:
I am not sure about improved readability and even more critical is the fact that in current implementation if the condition is met the loop braks, but if the condition is not met another loop round takes place (conversely, asserting would result in test failure.
I'd keep this part as it is, unless also another team member suggests to modify as per your comment.